spec.md

```diff
@@ -26,6 +27,61 @@ Enter CloudEvents, a specification for describing event data in a common way.
 CloudEvents seeks to ease event declaration and delivery across services,
 platforms and beyond.
 
+# Design Goals
+
+CloudEvents are typically used in a distributed system to allow for services to
+be loosely coupled during development, then may be configured with
+tight-coupling for specific applications or dynamically configured at runtime.
+
+The goal of the CloudEvents specification is to define interoperability of event
+systems that allow cloud providers, third-party tools and services to provide:
+- new event sources
+- new event consumers
+- observability, such as monitoring, debugging and visualization,
+- developer tools and libraries
+
+## Key properties of the system:
+- Source & action may be developed and deployed independently
+- Trigger configuration allows for actions to be invoked based on a condition
+(e.g. for a specific event type generated by a source, optionally filtered by
+specific attribute values, such as a Function which is invoked when an image is
+uploaded to specific S3 or Google Cloud Storage “bucket” path)
+- Events are uniquely identified to facilitate developer velocity and tooling:
+  - a unique idenfier allows for idempotency for reliable implementation
+  - structured event definitions allow the consumer to rely on a specific format
+- Common metadata attributes allow for observability via tools whithout
+knowledge of the specific implementation of an event system and also can provide
+value without detailed handling of the specifics of the unique event data
+- CloudEvents may be transported from source to action with different protocols.
+
+In order to achieve the above goals and desired system properties, the
+Serverless WG must agree on a common vocabulary with precise definitions
+describing the key elements of the system including:
+- [Source](../spec.md#source)
+- Action: An action consumes an event, producing a behavior or effect which was
+triggered by a specific *occurrence* from a specific *source*.  While the
+details of how events are delivered to an action is outside of the scope of the
+specification, the purpose of generating an *event* is typcially to allow other
+systems to easily react to changes in a source that they do not control. The
+*source* and action are often built by different developers.
+
+Examples:
+- the *source* is a managed service and the *action* is custom
+code in a serverless Function (such as AWS Lambda or Google Cloud Functions).
```

Comment:
let's leave the product names out - just remove the parens stuff - I don't think anything is lost by doing so.

Reply:
That sounds fine; I'll make this change.